### PR TITLE
Top nav breadcrumb

### PR DESCRIPTION
**Pull request by npfitz**
Epic #22826 

Milestone 1, step 1

This PR will move the collection breadcrumbs up to the app bar for questions. The new `PathBreadcrumbs` component was taken from [here](https://github.com/metabase/metabase/pull/19911/files#diff-86d61b50b806c202a8371d9ad39e14eaf9cf62327ebbfcabdea915c66b68dae4) @akiselev. Note that this makes a bit of a mess in the mobile view, which will be addressed after we spend some time with the new changes (@mazameli).

HupY1Fjt2s

### CODE REVIEW

File: frontend/src/metabase/nav/components/PathBreadcrumbs/PathBreadcrumbs.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      />
+    </PathContainer>
+  );
+};
+
+const propTypes = {
+  collection: PropTypes.object,
+};
+
+PathBreadcrumbs.propTypes = propTypes;
```
Comment (kulyk) [inline frontend/src/metabase/nav/components/PathBreadcrumbs/PathBreadcrumbs.tsx:81]: I think we prefer not to set prop types in typescript files as static type checking should be enough

File: frontend/src/metabase/nav/components/PathBreadcrumbs/PathBreadcrumbs.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        <Separator onClick={toggle} />
+      </>
+    ));
+  }
+  return (
+    <PathContainer>
+      {content}
+      <CollectionBadge
+        collectionId={collection.id}
+        inactiveColor={"text-medium"}
```
Comment (kulyk) [inline frontend/src/metabase/nav/components/PathBreadcrumbs/PathBreadcrumbs.tsx:71]: nit: you don't need brackets for string props: `inactiveColor="text-medium"`

File: frontend/src/metabase/nav/components/PathBreadcrumbs/PathBreadcrumbs.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+import {
+  PathSeparator,
+  PathContainer,
+  ExpandButton,
+} from "./PathBreadcrumbs.styled";
+
+interface Props {
+  collection: CollectionType;
+  collectionId: string;
```
Comment (kulyk) [inline frontend/src/metabase/nav/components/PathBreadcrumbs/PathBreadcrumbs.tsx:19]: You can also use `CollectionId` type from `metabase-types/api/collection`. Collection ID can be an integer or `null` / `"root"` string for "Our analytics"

File: frontend/src/metabase/nav/components/PathBreadcrumbs/PathBreadcrumbs.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+const PathBreadcrumbs = ({ collection }: Props) => {
+  const [isExpanded, setIsExpanded] = useState(false);
+  if (!collection) {
+    return null;
+  }
+  const toggle = () => setIsExpanded(!isExpanded);
```
Comment (kulyk) [inline frontend/src/metabase/nav/components/PathBreadcrumbs/PathBreadcrumbs.tsx:27]: opt: there is a [`useToggle` hook](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/hooks/use-toggle.ts) you can use here if you like it :)

**Review by kulyk — COMMENTED** [review]
Looking good! 😍 Left a few small comments

File: frontend/src/metabase/nav/components/PathBreadcrumbs/PathBreadcrumbs.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+          collectionId={parts[0].id}
+          inactiveColor={"text-medium"}
+        />
+        <Separator onClick={toggle} />
+        <ExpandButton
+          small
+          borderless
+          iconSize={10}
+          icon="ellipsis"
+          className={cx("text-dark")}
```
Comment (kulyk) [inline frontend/src/metabase/nav/components/PathBreadcrumbs/PathBreadcrumbs.tsx:47]: Guess there is no need in using `cx` here

File: frontend/src/metabase/nav/components/PathBreadcrumbs/PathBreadcrumbs.tsx (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/nav/components/PathBreadcrumbs/PathBreadcrumbs.tsx:81]: +1

File: frontend/src/metabase/nav/components/PathBreadcrumbs/PathBreadcrumbs.tsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+};
+
+const propTypes = {
+  collection: PropTypes.object,
+};
+
+PathBreadcrumbs.propTypes = propTypes;
+
+export default Collection.load({
+  id: (_state: EntitiesState, props: Props) => props.collectionId || "root",
```
Comment (ranquild) [inline frontend/src/metabase/nav/components/PathBreadcrumbs/PathBreadcrumbs.tsx:84]: `state` should be just `State` from `metabase-types/store`

File: frontend/src/metabase/redux/app.js [hunk 7]
```diff
@@ -72,7 +72,30 @@ const isNavbarOpen = handleActions(
   checkIsSidebarInitiallyOpen(),
 );
 
+export const SET_COLLECTION_ID = "metabase/app/SET_COLLECTION_ID";
+export const CLEAR_BREADCRUMBS = "metabase/app/CLEAR_BREADCRUMBS";
+export const setCollectionId = createAction(SET_COLLECTION_ID);
+export const clearBreadcrumbs = createAction(CLEAR_BREADCRUMBS);
+export const getBreadcrumbCollectionId = state =>
```
Comment (ranquild) [inline frontend/src/metabase/redux/app.js:79]: Other selectors for `app` are here `metabase/selectors.app.ts`. Should we move these new ones too?

File: frontend/src/metabase/redux/app.js [hunk 8]
```diff
@@ -72,7 +72,30 @@ const isNavbarOpen = handleActions(
...
+  collectionId: "",
+  show: false,
+};
+
+const breadcrumbs = handleActions(
+  {
+    [SET_COLLECTION_ID]: {
+      next: (state, { payload }) => ({ show: true, collectionId: payload }),
+    },
+    [CLEAR_BREADCRUMBS]: { next: () => ({ show: false, collectionId: "" }) },
```
Comment (ranquild) [inline frontend/src/metabase/redux/app.js:92]: Would it be possible to set `collectionId` to undefined or null instead?

**Review by ranquild — CHANGES_REQUESTED** [review]
There is an issue with Models, please double check if that's intended behavior

Screenshot 2022-05-26 at 15 54 56

File: frontend/src/metabase/redux/app.js (hunk 7 above)
Comment (npfitz) [reply frontend/src/metabase/redux/app.js:79]: Whoops! I just noticed "getIsNavbarOpen" in this file so I followed that pattern haha

**Comment by npfitz** [comment]
> There is an issue with Models, please double check if that's intended behavior
> 
> Screenshot 2022-05-26 at 15 54 56

Future work will be changing how the name of the model is shown, but currently we're focusing on the question view. Things are going to look a little weird over the next couple weeks
image

File: frontend/src/metabase-types/store/app.ts [hunk 9]
```diff
@@ -7,7 +9,13 @@ export interface AppErrorDescriptor {
   context?: string;
 }
 
+export interface AppBreadCrumbs {
```
Comment (ranquild) [inline frontend/src/metabase-types/store/app.ts:12]: @npfitz You also need to add `createMock` functions for the mocked store state. They are in the `mocks` folder here

**Review by ranquild — APPROVED** [review]
@npfitz LGTM

**Pull request merged**